ql/src/java/org/apache/hadoop/hive/ql/exec/DDLPlanUtils.java

```diff
@@ -0,0 +1,1005 @@
+  }
+
+  public void addDecimalStats(ColumnStatisticsData cd, List<String> ls) {
+    if (!cd.isSetDecimalStats()) {
+      return;
+    }
+    DecimalColumnStatsData dc = cd.getDecimalStats();
+    if(dc.isSetHighValue()) {
+      byte[] highValArr = setByteArrayToLongSize(dc.getHighValue().getUnscaled());
+      ls.add(highValue + ByteBuffer.wrap(highValArr).getLong() + "E" + dc.getHighValue().getScale() + "'");
+    }
+    if(dc.isSetLowValue()) {
+      byte[] lowValArr = setByteArrayToLongSize(dc.getLowValue().getUnscaled());
+      ls.add(lowValue + ByteBuffer.wrap(lowValArr).getLong() + "E" + dc.getLowValue().getScale() + "'");
+    }
+    ls.add(numNulls + dc.getNumNulls() + "'");
+    ls.add(numDVs + dc.getNumDVs() + "'");
+  }
+
+  public void addDoubleStats(ColumnStatisticsData cd, List<String> ls) {
+    if (!cd.isSetDoubleStats()) {
+      return;
+    }
+    DoubleColumnStatsData dc = cd.getDoubleStats();
+    ls.add(numNulls + dc.getNumNulls() + "'");
+    ls.add(numDVs + dc.getNumDVs() + "'");
+    ls.add(highValue + dc.getHighValue() + "'");
+    ls.add(lowValue + dc.getLowValue() + "'");
+  }
+
+  public String checkBitVectors(ColumnStatisticsData cd) {
+    if (cd.isSetDoubleStats() && cd.getDoubleStats().isSetBitVectors()) {
+      return Base64.getEncoder().encodeToString(cd.getDoubleStats().getBitVectors());
+    }
+    if (cd.isSetBinaryStats() && cd.getBinaryStats().isSetBitVectors()) {
+      return Base64.getEncoder().encodeToString(cd.getBinaryStats().getBitVectors());
+    }
+    if (cd.isSetStringStats() && cd.getStringStats().isSetBitVectors()) {
+      return Base64.getEncoder().encodeToString(cd.getStringStats().getBitVectors());
+    }
+    if (cd.isSetDateStats() && cd.getDateStats().isSetBitVectors()) {
+      return Base64.getEncoder().encodeToString(cd.getDateStats().getBitVectors());
+    }
+    if (cd.isSetLongStats() && cd.getLongStats().isSetBitVectors()) {
+      return Base64.getEncoder().encodeToString(cd.getLongStats().getBitVectors());
+    }
+    if (cd.isSetDecimalStats() && cd.getDecimalStats().isSetBitVectors()) {
+      return Base64.getEncoder().encodeToString(cd.getDecimalStats().getBitVectors());
+    }
+    if (cd.isSetDoubleStats() && cd.getBooleanStats().isSetBitVectors()) {
+      return Base64.getEncoder().encodeToString(cd.getBooleanStats().getBitVectors());
+    }
+    return null;
+  }
+
+  /**
+   * Parses the basic ColumnStatObject and returns the alter tables stmt for each individual column in a table.
+   *
+   * @param columnStatisticsData
+   * @param colName
+   * @param tblName
+   * @param dbName
+   * @return
+   */
+  public String getAlterTableStmtCol(ColumnStatisticsData columnStatisticsData, String colName, String tblName, String dbName) {
+    ST command = new ST(ALTER_TABLE_UPDATE_STATISTICS_TABLE_COLUMN);
+    command.add(DATABASE_NAME, dbName);
+    command.add(TABLE_NAME, tblName);
+    command.add(COLUMN_NAME, colName);
+    List<String> temp = new ArrayList<>();
```

Comment:
Is this code getting duplicated with "getAlterTableStmtPartitionColStat". Extract to a new method?